test/e2e_node/image_conformance_test.go

```diff
@@ -0,0 +1,156 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e_node
+
+import (
+	"time"
+
+	"k8s.io/kubernetes/pkg/client/restclient"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	imageRetryTimeout = time.Minute * 2
+	imagePullInterval = time.Second * 15
+)
+
+type ConformanceTestPhase string
+
+const (
+	ConformanceTestTargetNew   ConformanceTestPhase = "new"
+	ConformanceTestTargetReady ConformanceTestPhase = "ready"
+	ConformanceTestTargetFree  ConformanceTestPhase = "free"
+	ConformanceTestTargetError ConformanceTestPhase = "error"
+)
+
+var _ = Describe("Image Conformance Test", func() {
+	var cl *client.Client
+
+	BeforeEach(func() {
+		// Setup the apiserver client
+		cl = client.NewOrDie(&restclient.Config{Host: *apiServerAddress})
+	})
+
+	Describe("image conformance blackbox test", func() {
+		type ImageCase struct {
+			Image     ConformanceImage
+			Phase     ConformanceTestPhase
+			PullError error
+		}
+		Context("when testing images that exist", func() {
+			var imageCases []ImageCase
+			BeforeEach(func() {
+				imageExistTags := []string{
+					"gcr.io/google_containers/busybox:latest",
+					"gcr.io/google_containers/mounttest:latest",
+				}
+				for _, imageExistTag := range imageExistTags {
+					var imageCase ImageCase
+					imageExist, _ := NewConformanceImage("docker", imageExistTag)
+					imageCase.Image = imageExist
+					imageCase.Phase = ConformanceTestTargetNew
+					// Pulling images from gcr.io is flaky, so retry failures
+					for start := time.Now(); time.Since(start) < imageRetryTimeout; time.Sleep(imagePullInterval) {
+						if imageCase.PullError = imageCase.Image.Pull(); imageCase.PullError == nil {
+							imageCase.Phase = ConformanceTestTargetReady
```

Comment:
@liangchenye Removing image twice seems to be acceptable. :) We usually do that in e2e test, such as:
```
defer DeletePod()

// do something

DeletePod()
```